File: CajeroAngular/src/app/models/tarjeta/tarjeta.ts
export class Tarjeta{
    id:number;
    numeroTarjeta:string;
    pin:number;
    balance:Float32Array;
    fechaVencimiento:Date;
    bloqueada:boolean;

    constructor(id:number,numeroTarjeta:string,pin:number,balance:Float32Array,fechaVencimiento:Date,bloqueada:boolean){
        this.id = id;
        this.numeroTarjeta = numeroTarjeta;
        this.pin = pin;
        this.balance = balance;
        this.fechaVencimiento = fechaVencimiento;
        this.bloqueada = bloqueada;
    }
}

File: CajeroApi/CajeroApi/Models/Tarjeta.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CajeroApi.Models
{
    public class Tarjeta
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string NumeroTarjeta { get; set; }
        [Required]
        public int Pin { get; set; }
        public float Balance { get; set; }
        public DateTime FechaVencimiento { get; set; }

        public bool Bloqueada { get; set; }
    }
}
